src/model/setup.jl

```diff
@@ -161,6 +165,34 @@ function add_optimizer!(config, data, model)
     set_optimizer(model, optimizer_factory; add_bridges=false)
 end
 
+"""
+    create_capex_obj!(config, data) -> 
+
+Creates capex_obj and transmission_capex_obj columns which are the capex cost seen in the objective function. It is a ByYear column that is only non zero for year_on.
+Set to capex for unbuilt generators in and after the year_on
+Set to 0 for already built capacity because capacity expansion isn't considered for existing generators  
+"""
+function create_capex_obj!(config, data)
+    gen = get_table(data, :gen)
+    years = get_years(data)
+
+    #warn if capex_obj already exists
+    :capex_obj in propertynames(data[:gen]) && @warn "capex_obj hasn't been calculated yet but appears in the gen table. It will be overwritten."
+
+    capex_obj = Container[ByNothing(0.0) for i in 1:nrow(gen)]
+    transmission_capex_obj = Container[ByNothing(0.0) for i in 1:nrow(gen)]
+    add_table_col!(data, :gen, :capex_obj, capex_obj, DollarsPerMWBuiltCapacityPerHour, "Hourly capital expenditures that is passed into the objective function. 0 for already built capacity")
+    add_table_col!(data, :gen, :transmission_capex_obj, transmission_capex_obj, DollarsPerMWBuiltCapacityPerHour, "Hourly capital expenditures for transmission that is passed into the objective function. 0 for already built capacity")
+
+
+    for (idx_g, g) in enumerate(eachrow(gen))
+        capex_years = filter(year -> year >= g.year_on && year < add_to_year(g.year_on, g.econ_life), years)
```

Comment:
Good idea multiplying by a binary ByYear.

You could probably allocate a bit less by doing something like:
```julia
capex_filter = ByYear(map(year -> year >= g.year_on && year < add_to_year(g.year_on, g.econ_life), years))
g.capex_obj = g.capex .* capex_filter
```